torchvision/datasets/utils.py

```diff
@@ -384,10 +395,14 @@ def _decompress(from_path: str, to_path: Optional[str] = None, remove_finished:
     if remove_finished:
         os.remove(from_path)
 
-    return to_path
+    return pathlib.Path(to_path)
 
 
-def extract_archive(from_path: str, to_path: Optional[str] = None, remove_finished: bool = False) -> str:
+def extract_archive(
+    from_path: Union[str, pathlib.Path],
+    to_path: Optional[Union[str, pathlib.Path]] = None,
+    remove_finished: bool = False,
+) -> pathlib.Path:
```

Comment:
Unfortunately, forcing the return type to be a `Path` when it previously was a `str` is a BC-breaking change. It doesn't really matter for `_decompress` because it's private (leading underscore), but it's an issue here with `extract_archive` which is technically public.

(Side note: we didn't intend for any of the stuff in `datasets/utils.py` to ever be public, but that's another story).

We have 2 options here:

- always return a `str`. This is sub-ideal but simpler. And since we never intended those utils to be public anyway, it's kinda OK - we should deprecate them and make them private eventually, but that's for another day.
- always return the same type as `from_path`. This is the best UX, and requires a bit more logic.

Reply:
Great point. I have done the second option (though technically speaking that also changes behavior from the public API).